servers/rendering/rendering_device.cpp

```diff
@@ -4236,6 +4317,83 @@ void RenderingDevice::draw_list_bind_index_array(DrawListID p_list, RID p_index_
 	}
 }
 
+RID RenderingDevice::blas_create(RID p_vertex_array, RID p_index_array, RID p_transform_buffer, uint64_t p_transform_offset) {
+	ERR_FAIL_COND_V_MSG(!has_feature(SUPPORTS_RAYTRACING), RID(), "The current rendering device has no raytracing support.");
+
+	VertexArray *vertex_array = vertex_array_owner.get_or_null(p_vertex_array);
+	ERR_FAIL_NULL_V(vertex_array, RID());
+	RDD::VertexFormatID vertex_format;
+	if (vertex_array->description != INVALID_ID) {
+		ERR_FAIL_COND_V(!vertex_formats.has(vertex_array->description), RID());
+		vertex_format = vertex_formats[vertex_array->description].driver_id;
+	}
+	_check_transfer_worker_vertex_array(vertex_array);
+
+	// Indices are optional.
+	IndexArray *index_array = index_array_owner.get_or_null(p_index_array);
+	RDD::BufferID index_buffer = RDD::BufferID();
+	IndexBufferFormat index_format = IndexBufferFormat::INDEX_BUFFER_FORMAT_UINT32;
+	uint32_t index_offset_bytes = 0;
+	uint32_t index_count = 0;
+	if (index_array) {
+		index_buffer = index_array->driver_id;
+		index_format = index_array->format;
+		index_offset_bytes = index_array->offset * (index_array->format == INDEX_BUFFER_FORMAT_UINT16 ? sizeof(uint16_t) : sizeof(uint32_t));
+		index_count = index_array->indices;
+		_check_transfer_worker_index_array(index_array);
+	}
+
+	Buffer *transform_buffer = storage_buffer_owner.get_or_null(p_transform_buffer);
+	ERR_FAIL_NULL_V(transform_buffer, RID());
+	ERR_FAIL_COND_V_MSG(!transform_buffer->usage.has_flag(RDD::BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT), RID(), "Transform buffer provided was not created for shader device address usage.");
+	ERR_FAIL_COND_V_MSG(!transform_buffer->usage.has_flag(RDD::BUFFER_USAGE_ACCELERATION_STRUCTURE_BUILD_INPUT_READ_ONLY_BIT), RID(), "Transform buffer provided was not created for acceleration structure build input.");
+	_check_transfer_worker_buffer(transform_buffer);
+
+	AccelerationStructure acceleration_structure;
+	acceleration_structure.type = RDD::ACCELERATION_STRUCTURE_TYPE_BLAS;
+	acceleration_structure.driver_id = driver->blas_create(vertex_array->buffers[0], vertex_array->offsets[0], vertex_format, vertex_array->vertex_count, index_buffer, index_format, index_offset_bytes, index_count, transform_buffer->driver_id, p_transform_offset);
+	ERR_FAIL_COND_V(!acceleration_structure.driver_id, RID());
+
+	acceleration_structure.draw_tracker = RDG::resource_tracker_create();
+	acceleration_structure.draw_tracker->acceleration_structure_driver_id = acceleration_structure.driver_id;
```

Comment:
I am not sure if this is the right function for tracking logic. Maybe `raytracing_list_build_acceleration_structure` would be a better choice?

Also, in `rendering_device_graph.cpp`, I am using this `acceleration_structure_driver_id` to create a barrier before accessing if from raytracing shaders, but I am already doing the same thing explicitly in `_run_raytracing_list_command` while handling the `RaytracingListInstruction::TYPE_BUILD_ACCELERATION_STRUCTURE`. 🤔

Reply:
@DarioSamo, as you suggested, I introduced a new resource usage (`ACCELERATION_STRUCTURE_BUILD_INPUT`).
Then, I had to update the demo and the renderer branch splitting the raytracing list in 3 parts: 
1. Build BLASs
2. Build TLAS
3. Trace rays

So 1 generates inputs for 2, and 2 generate inputs for 3, while proper memory barriers are inferred. With this, I removed the explicit barriers mentioned in the comment above.

Reply:
@DarioSamo, new version, I guess this fits better in the draw graph. I introduced a new build acceleration command, as you suggested, separated from the "list".

Reply:
Seems good, although TYPE_BUILD_ACCELERATION_STRUCTURE seems to remain on RaytracingListInstruction and is probably meant to be removed.